client/components/category-pill-navigation/index.tsx

```diff
@@ -70,6 +75,53 @@ export const CategoryPillNavigation = ( {
 		} );
 	}, [ selectedCategoryId ] );
 
+	if ( isMobile ) {
+		const currentUrl =
+			typeof window === 'undefined' ? '' : window.location.pathname + window.location.search;
+
+		const selectedText = [ ...categories, ...( buttons || [] ) ].find(
+			( { link } ) => currentUrl === addLocaleToPathLocaleInFront( link )
+		)?.label;
+
+		return (
+			<div className="category-pill-navigation">
+				<SelectDropdown
+					className="category-pill-navigation__mobile-select"
+					selectedText={ selectedText }
+				>
+					{ buttons &&
+						buttons.map( ( button ) => {
+							const value = addLocaleToPathLocaleInFront( button.link );
+
+							return (
+								<SelectDropdown.Item
+									key={ button.label }
+									selected={ value === currentUrl }
+									onClick={ () => page( value ) }
+								>
+									{ button.label }
+								</SelectDropdown.Item>
+							);
+						} ) }
+					{ categories &&
+						categories.map( ( category ) => {
+							const value = addLocaleToPathLocaleInFront( category.link );
+
+							return (
+								<SelectDropdown.Item
+									key={ category.id }
+									selected={ value === currentUrl }
+									onClick={ () => page( value ) }
+								>
+									{ category.label }
+								</SelectDropdown.Item>
+							);
+						} ) }
+				</SelectDropdown>
+			</div>
+		);
```

Comment:
```suggestion
		const selectedItem =
			buttons?.find( ( { isActive } ) => isActive ) ||
			categories.find( ( { id } ) => id === selectedCategoryId );

		return (
			<div className="category-pill-navigation">
				<SelectDropdown
					className="category-pill-navigation__mobile-select"
					selectedText={ selectedItem?.label }
				>
					{ buttons?.map( ( button ) => (
						<SelectDropdown.Item
							key={ button.label }
							path={ addLocaleToPathLocaleInFront( button.link, locale ) }
							selected={ button.isActive }
						>
							{ button.label }
						</SelectDropdown.Item>
					) ) }

					{ categories?.map( ( category ) => (
						<SelectDropdown.Item
							key={ category.id }
							path={ addLocaleToPathLocaleInFront( category.link, locale ) }
							selected={ category.id === selectedCategoryId }
						>
							{ category.label }
						</SelectDropdown.Item>
					) ) }
				</SelectDropdown>
			</div>
		);
```
A few things are going on here:
- I think we ought to use `button.isActive` and `selectedCategoryId` to determine which nav item is active. I.e. copy the logic we use for the desktop nav.
- `SelectDropdown.Item` always render as `<a>` elements. The `path` prop becomes an `href` on the link. With this approach, we don't get the `isLoggedIn` logic of `LocalizedLink`, but I think that's OK.
- We don't need the `button &&` and `categories &&` checks if we use optional chaining (i.e., `buttons?.map` and `categories?.map`

Reply:
Good points 👍
And regarding `locale` as second argument to `addLocaleToPathLocaleInFront`, I am thinking - does it make sense to provide it from useLocale hook?

Reply:
Oh yeah, let's use the `useLocale` hook for that 👍